src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DefaultValidationMetadataProvider.cs

```diff
@@ -12,15 +11,24 @@ namespace Microsoft.AspNet.Mvc.ModelBinding.Metadata
     /// </summary>
     public class DefaultValidationMetadataProvider : IValidationMetadataProvider
     {
+        private readonly ValidationErrorMetadata _validationErrorMetadata;
+
+        public DefaultValidationMetadataProvider(ValidationErrorMetadata validationErrorMetadata)
+        {
+            _validationErrorMetadata = validationErrorMetadata;
+        }
+
         /// <inheritdoc />
         public void GetValidationMetadata([NotNull] ValidationMetadataProviderContext context)
         {
+            context.ValidationMetadata.ValidationErrorMetadata = _validationErrorMetadata;
```

Comment:
I'd just create the copy. It's an extra object, but it happens once per model and it's cached.